crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -2125,7 +2127,7 @@ async fn test_custom_property_parse_published_at() {
 
 #[tokio::test]
 #[cfg_attr(msim, ignore)]
-async fn test_custom_property_ensure_published_at() {
+async fn test_custom_property_unpublished_dependencies_error() {
```

Comment:
Doing a stock take of all our tests -- do we have tests for all the following cases?

- Publishing with dependencies (which therefore need `published-at` set) -- I think the answer is yes because of `Sui` and `MoveStdlib`.
- Publishing and there are either missing deps and/or unparseable deps (this is similar to the test that I'm commenting on, but actually tries to go through the publish flow, not just error generation).
- Publishing with unpublished dependencies (again, I think yes, above).
- Publishing with unpublished dependencies, but one of them has an unparseable address (this should fail, and I think maybe it doesn't today, see my comment below).

Reply:
> Publishing and there are either missing deps and/or unparseable deps (this is similar to the test that I'm commenting on, but actually tries to go through the publish flow, not just error generation).

Right, there's a `publish` simtest that exercises the good path for the actual publish flow. Let me see about adding one for missing / invalid deps (am a tad hesitant with simtest because of potential flakey-ness, but for publish we do want some assurances here).

Reply:
(should be possible to sidestep simtest if we're only testing the error cases through that codepath)

Reply:
FWIW it is not simtest itself that is flaky.  There are some tests running on simtest that are flaky.  In general, simtest makes non-deterministic tests deterministic and that is working fine, so if you were to add a new simtest and it was going to work before, it would work now as well.

Reply:
Needed to add simtest because doesn't seem simple to set up `context` and other values going through the command.

> Publishing with unpublished dependencies, but one of them has an unparseable address 

added an explicit simtest for this.

> Publishing and there are either missing deps and/or unparseable deps

added an explicit simtest for this too (the missing deps case. The unparseable + missing is covered by above, since we always fail for unparseable case now).

These simtest additions may seem like they subsume the unit tests, but given simtest flakiness I'd like to keep all the tests and monitor simtest stability for these new ones in the next couple of days.

Reply:
> There are some tests running on simtest that are flaky.

Yep, this is what I'm trying to get at, sorry that wasn't clear. I don't have a great grasp of whether some of these are flaky, but I did hints around this, like a filter on the `test_package_publish_command` I was dealing with:

https://github.com/MystenLabs/sui/blob/009e21066b99955aad2bf1a9ff5468d05eeef9e9/scripts/simtest/simtest-run.sh#L9-L10

I'm not sure if this filter is applied in CI or elsewhere, it _does_ run locally. So this just makes me conscious / cautious of the possibility of flakiness around this functionality.